memory_map.py

```diff
@@ -4,8 +4,44 @@
 
 """Functions for converting Devicetrees to the template parameterization"""
 
+import re
 import sys
 
+def get_ram_memories(tree):
+    """Given a Devicetree, get the list of ram memories to describe in the
+       linker script"""
+
+    # RAMs (TIM, LIM, ILS, DLS, main memory) that may have ECC protection
+    # define ram list filters
+    itim = re.compile("itim")
+    dtim = re.compile("dtim")
+    sram = re.compile("sys-sram")
+    ils = re.compile("ils")
+    dls = re.compile("dls")
+    memory = re.compile("memory")
+
+    memories = dict()
+    # Get a list of RAM nodes with node as the its key
+    nodes = list(tree.all_nodes())
+    for node in nodes:
+        if any([itim.search(node.name), dtim.search(node.name), \
+                sram.search(node.name), ils.search(node.name),  \
+                dls.search(node.name), memory.search(node.name)]):
+
+            region = dict()
+            region.update(name=node.name, node=node, path=node.get_path(), \
+                          region=0, offset=0)
+            memories.update({node.name : region})
+
+    if len(memories) == 0:
+        print("ERROR: No RAM is not defined by the Devicetree")
```

Comment:
Double-negative "no ram is not" should be replaced with a single-negative.

Reply:
> Double-negative "no ram is not" should be replaced with a single-negative.

Ahh, thanks will fix